mevcommitclient/mev_commit_client.go

```diff
@@ -0,0 +1,136 @@
+package mevcommitclient
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	builderRegistry "github.com/primev/mev-commit/contracts-abi/clients/ProviderRegistry"
+	validatoroptinrouter "github.com/primev/mev-commit/contracts-abi/clients/ValidatorOptInRouter"
+)
+
+type BuilderRegisteredEvent struct {
+	Builder      common.Address
+	Value        *big.Int
+	BlsPublicKey []byte
+	BlockNumber  uint64
+	TxHash       common.Hash
+	Timestamp    time.Time
+	IsValid      bool
+}
+
+type IMevCommitClient interface {
+	GetOptInStatusForValidators(pubkeys [][]byte) ([]bool, error)
+	GetActiveBuilders() ([][]byte, error)
+}
+
+type MevCommitClient struct {
+	L1Address                  string
+	MevCommitAddress           string
+	ValidatorRouterAddress     common.Address
+	BuilderRegistryAddress     common.Address
```

Comment:
Here and everywhere else in the file, to avoid confusion we should use the name of the contract from source code, "provider registry"